### PR TITLE
Feature/63 get length

### PR DESCRIPTION
**Pull request by StokesMIDE**
This PR adds an improved (read: functional) recording file interval calculator, as well as refactors the `getExitCondition()` function to work with non-file streams.

The `exitCondition` argument was removed from the `importer` functions and `Dataset.__init__()` because it gets read during the import of the file, anyway.

### CODE REVIEW

**Review by CFlaniganMide — APPROVED** [review]
Looks good, can't find anything to change.

File: idelib/attributes.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    for d in data:
+        if isinstance(d[1], (tuple, list)) and not isinstance(d[1], time.struct_time):
+            k = d[0]
+            v, elementType = d[1]
+        else:
+            k, v = d
+            for t, elementType in attTypes:
+                if isinstance(v, t):
+                    break
+                elementType = 'BinaryAttribute'
```
[the remark this answers is not in the document]
Comment (StokesMIDE) [reply idelib/attributes.py]: The `attributes.py` code is just what was in `util.py`, unmodified; for this PR, I'd rather leave as much of the existing code as-is as I can. It's an attempt to reign in my tendency to scope-creep.

[the remark this answers is not in the document]
Comment (CFlaniganMide) [reply idelib/attributes.py]: yeah, I got mixed up about what code was from what PR

**Comment by StokesMIDE** [comment]
This has gotten one approval; if there are no more reviews, it will be merged shortly.

**Pull request merged**